game/src/main/java/org/apollo/game/sync/task/PrePlayerSynchronizationTask.java

```diff
@@ -58,6 +60,15 @@ public void run() {
 		Position old = player.getPosition();
 		player.getWalkingQueue().pulse();
 
+		if (player.getSecondDirection() != Direction.NONE) {
```

Comment:
This isn't right. We should only decrement the run energy when the actually move. The direction could be set but the collision maps said this is an invalid move.

Reply:
I wasn't fond of the condition, but it seemed to be the best fitting choice for what needed to be done. Sorry if I'm wrong here and just wasting your time, but I believe the direction is only set if the collision map allows the movement (see https://github.com/apollo-rsps/apollo/blob/kotlin-experiments/game/src/main/java/org/apollo/game/model/entity/WalkingQueue.java#L156). 

Also, am I wrong to assume that if I use ``WalkingQueue.isRunning()`` & ``WalkingQueue.size()``, _then_ I would run into the issue that you warned about? I believe when adding steps into the walking queue, it won't check for collision.

Reply:
Hmm, I need to re-look at the assignment of `secondDirection`, you may be right there.

Reply:
@garyttierney any update on this?

Reply:
The code is fine as is, and what we expect. That's the only assignment of `secondDirection` and the point @Tomm0017 made is valid.